lib/imagery/list_gp.c

```diff
@@ -36,7 +36,7 @@ int I_list_group(const char *group, const struct Ref *ref, FILE * fd)
     }
     max = 0;
     for (i = 0; i < ref->nfiles; i++) {
-	sprintf(buf, "<%s@%s>", ref->file[i].name, ref->file[i].mapset);
+        sprintf(buf, "<%.38s@%.38s>", ref->file[i].name, ref->file[i].mapset); /* truncates! */
```

Comment:
Use
```
char *buf = NULL;
...
G_asprintf(&buf, "<%s@%s>", ref->file[i].name, ref->file[i].mapset);
```
This is in a for loop, thus `G_free()` would need to be added.

Reply:
The problem with these `I_list_group()` functions of the  `lib/imagery/list_[...].c` files, is that they result in "nicely" formatted strings, fitting in a 80 column. My solution is probably not the best either, as there is no way to know if truncated or not. In addition, there are two loops (per function), one is for calculating width.

Perhaps I can come up with a solution with added ellipsis in case of truncation?

Reply:
Also, my present solution is limited by truncating mapname *and* location separately. Instead their combined length should be accounted for.

Reply:
These (what is printed to buf) are names, thus `GNAME_MAX + GNAME_MAX + 1` (+ 1 for the `@`) might work?
Truncating means removing information and creating ambiguities.

Reply:
Just pushed suggestion of an implementation that truncates when needed.
Example of its results with various input:
```
-------------------------------------------------------------------------------| <- 80 char
# Long map name:
ifonlythisisalandsatfilewithaveryverylongnameandagainifonlythisisalandsatfilewithaveryverylongnamePERMANENT
<ifonlythisisalandsatfilewithaveryverylongnameandagainifonly...@PERMANENT>    

# Long mapset name:
shortname@IFONLYTHISISALANDSATFILEWITHAVERYVERYLONGNAMEANDAGAINIFONLYTHISISALANDSATFILEWITHAVERYVERYLONGNAME
<shortname@IFONLYTHISISALANDSATFILEWITHAVERYVERYLONGNAMEANDAGAINIFONLY...>    

# Long map and mapset name:
ifonlythisisalandsatfilewithaveryverylongnameandagainifonlythisisalandsatfilewithaveryverylongname@IFONLYTHISISALANDSATFILEWITHAVERYVERYLONGNAMEANDAGAINIFONLYTHISISALANDSATFILEWITHAVERYVERYLONGNAME
<ifonlythisisalandsatfilewithavery...@IFONLYTHISISALANDSATFILEWITHAVER...>    

# No need to truncate (no changed behaviour):
ifonlythisisalandsatfilewithavery@IFONLYTHISISALANDSATFILEWITHAVERY
<ifonlythisisalandsatfilewithavery@IFONLYTHISISALANDSATFILEWITHAVERY>    
-------------------------------------------------------------------------------| <- 80 char
```

Reply:
OK, I agree that such long names are probably rare, truncating in these cases should be harmless.